src/backup.sh

```diff
@@ -0,0 +1,399 @@
+                continue
+                ;;
+            esac
+          fi
+          cp "$day_file" "$KW_DATA_DIR/pomodoro/$year$month/$day"
+        done
+      done
+    done
+
+    if [[ -f "$decompress_path/pomodoro/tags" ]]; then
+      if [[ -f "$KW_DATA_DIR/pomodoro/tags" ]] &&
+        ! cmp -s "$KW_DATA_DIR/pomodoro/tags" "$decompress_path/pomodoro/tags"; then
+        if [[ -z "$decision" ]]; then
+          complain 'pomodoro/tags already exists'
+          complain 'Do you want to:'
+          say "(1) Replace it"
+          say "(2) Keep it"
+          say "(3) Aggregate it"
+          read -r decision
+        fi
+        case "$decision" in
+          1)
+            cp "$decompress_path/pomodoro/tags" "$KW_DATA_DIR/pomodoro/tags"
+            ;;
+          2)
+            return
+            ;;
+          3)
+            difference=$(diff -u "$KW_DATA_DIR/pomodoro/tags" "$decompress_path/pomodoro/tags")
+            patch "$KW_DATA_DIR/pomodoro/tags" <<< "$difference"
+            ;;
+        esac
+        return
+      fi
+      cp "$decompress_path/pomodoro/tags" "$KW_DATA_DIR/pomodoro/tags"
+    fi
+  fi
+}
+
+function restore_statistics()
+{
+  local decision
+  local year
+  local month
+  local day
+  local difference
+
+  if [[ -d "$decompress_path/statistics" ]]; then
+    for year_dir in "$decompress_path/statistics/"*/; do
+      year="$(str_remove_prefix "$year_dir" "$decompress_path/statistics/")"
+      mkdir -p "$KW_DATA_DIR/statistics/$year"
+      for month_dir in "$year_dir"*; do
+        month="$(str_remove_prefix "$month_dir" "$decompress_path/statistics/$year")"
+        mkdir -p "$KW_DATA_DIR/statistics/$year$month"
+        for day_file in "$month_dir"/*; do
+          day="$(str_remove_prefix "$day_file" "$decompress_path/statistics/$year$month/")"
+          if [[ -f "$KW_DATA_DIR/statistics/$year$month/$day" ]] &&
+            ! cmp -s "$KW_DATA_DIR/statistics/$year$month/$day" "$day_file"; then
+            if [[ -z "$decision" ]]; then
+              complain "We found a file from $year$month/$day in your kw statistics directory."
+              complain "Do you want to:"
+              say "(1) Replace all duplicate files with backup"
+              say "(2) Keep all the old files"
+              say "(3) Aggregate all old and backup files"
+              read -r decision
+            fi
+
+            case "$decision" in
+              1)
+                cp "$day_file" "$KW_DATA_DIR/statistics/$year$month/$day"
+                continue
+                ;;
+              2)
+                continue
+                ;;
+              3)
+                difference=$(diff -u "$KW_DATA_DIR/statistics/$year$month/$day" "$day_file")
+                patch "$KW_DATA_DIR/statistics/$year$month/$day" <<< "$difference"
+                continue
+                ;;
+            esac
+          fi
+          cp "$day_file" "$KW_DATA_DIR/statistics/$year$month/$day"
+        done
+      done
```

Comment:
I think you can try to create a function that encapsulates this idea since it is shared between restore statistics and Pomodoro.